### DISCUSSION TITLE
Time to add Double-ended queue?

### DISCUSSION THREAD

**Discussion by tasty0tomato** [dotnet/csharplang#5814] [Language Ideas] [answered]
There is no Double-ended queue in C# which may be kind of annoying...

**Comment by bernd5** [comment]
What do you mean by "Double-ended queue"?

**Reply by tasty0tomato** [reply]
Something like std::deque in C++
https://en.cppreference.com/w/cpp/container/deque

**Reply by bernd5** [reply]
Ah, I see. There are some implementations - just search on github or nuget. This one for [example](https://github.com/StephenCleary/Deque).
But this has nothing todo with the language...

**Comment by HaloFour** [comment, marked as answer]
API requests should be made to the [runtime repository](https://github.com/dotnet/runtime/).

**Reply by tasty0tomato** [reply]
Sorry, you are right.
https://github.com/dotnet/runtime/issues/935 maybe is what I need

**Comment by TonyValenti** [comment]
You should take a look at linked list. It is a deque that just doesn't have the nicely named methods.
